utils/cache_utils.py

```diff
@@ -154,6 +157,12 @@ def find_distance(
             self.cache[position_str][target_str] = native_distance_function(
                 position, target
             )
+            self.misses += 1
+        else:
+            self.hits += 1
+        self.num_accesses += 1
+        if self.num_accesses % 1000 == 0:
+            print("Cache Miss-Hit Ratio: %.4f" % (self.misses/self.hits))
```

Comment:
At most this should be included as debug information and not printed straight to the stdout.

Reply:
Agreed. In fact we might even remove it (or t least have it print less frequently)